### PR TITLE
Move storage.tsdb.retention.size out of experimental #8728

### PR DESCRIPTION
**Pull request by goelankitt**
Fixes https://github.com/prometheus/prometheus/issues/8728

### CODE REVIEW

**Comment by goelankitt** [comment]
@roidelapluie Are any other changes required?

**Comment by cstyan** [comment]
The DCO is missing: https://github.com/probot/dco#how-it-works 

You can `commit --amend` probably.

**Comment by goelankitt** [comment]
@cstyan Fixed it :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## 2.28.0 / 2021-06-21
 
+* [CHANGE] Move storage.tsdb.retention.size out of experimental #8728
```
Comment (roidelapluie) [inline CHANGELOG.md:3]: Please do not modify the changelog, especially since 2.28 has been released already.

Comment (goelankitt) [reply CHANGELOG.md]: Removed this change :)

**Comment by roidelapluie** [comment]
Thank you! Welcome to the community :)

File: cmd/prometheus/main.go [hunk 2]
```diff
@@ -258,7 +258,7 @@ func main() {
 	a.Flag("storage.tsdb.retention.time", "How long to retain samples in storage. When this flag is set it overrides \"storage.tsdb.retention\". If neither this flag nor \"storage.tsdb.retention\" nor \"storage.tsdb.retention.size\" is set, the retention time defaults to "+defaultRetentionString+". Units Supported: y, w, d, h, m, s, ms.").
 		SetValue(&newFlagRetentionDuration)
 
-	a.Flag("storage.tsdb.retention.size", "[EXPERIMENTAL] Maximum number of bytes that can be stored for blocks. A unit is required, supported units: B, KB, MB, GB, TB, PB, EB. Ex: \"512MB\". This flag is experimental and can be changed in future releases.").
+	a.Flag("storage.tsdb.retention.size", "Maximum number of bytes that can be stored for blocks. A unit is required, supported units: B, KB, MB, GB, TB, PB, EB. Ex: \"512MB\". This flag is experimental and can be changed in future releases.").
```
Comment (candlerb) [inline cmd/prometheus/main.go:261]: Although the `[EXPERIMENTAL]` tag at the front has gone, the end of the text still says:

> This flag is experimental and can be changed in future releases.

Comment (roidelapluie) [reply cmd/prometheus/main.go:261]: Thanks: https://github.com/prometheus/prometheus/pull/9140

**Pull request merged**